src/NextGen/DonationForm/Rules/SubscriptionInstallmentsRule.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Give\NextGen\DonationForm\Rules;
+
+use Closure;
+use Give\Donations\ValueObjects\DonationType;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidatesOnFrontEnd;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidationRule;
+
+class SubscriptionInstallmentsRule implements ValidationRule, ValidatesOnFrontEnd {
+
+    /**
+     * @unreleased
+     */
+    public static function id(): string
+    {
+        return 'subscriptionInstallments';
+    }
+
+    /**
+     * @unreleased
+     */
+    public static function fromString(string $options = null): ValidationRule
+    {
+        return new self();
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke($value, Closure $fail, string $key, array $values): bool
+    {
+        $donationType = new DonationType($values['donationType']);
+
+        return $donationType->isSubscription() && is_numeric($value);
```

Comment:
> The second condition refers to is_numeric($value), but $value (singular) doesn't seem to be defined. An array $values (plural) is an argument of the method - do you mean to validate that each value is numeric?

Same thing here.

Or perhaps the argument is wrong and should also be singular?

Reply:
Resolved in another conversation.